src/main/java/de/tum/cit/aet/artemis/communication/service/notifications/NotificationSettingsService.java

```diff
@@ -397,65 +398,53 @@ private Map<NotificationType, Boolean> convertNotificationSettingsToNotification
+     * @param user                     The user for whom the settings should be checked.
+     * @return The updated and correct notification settings.
      */
     public Set<NotificationSetting> checkLoadedNotificationSettingsForCorrectness(Set<NotificationSetting> userNotificationSettings, User user) {
+        var defaultSettings = DEFAULT_NOTIFICATION_SETTINGS;
         if (userNotificationSettings.isEmpty()) {
-            return DEFAULT_NOTIFICATION_SETTINGS;
+            return defaultSettings;
         }
-        // default settings might have changed (e.g. number of settings) -> need to merge the saved settings with default ones (else errors appear)
-
-        if (!compareTwoNotificationSettingsSetsBasedOnSettingsId(userNotificationSettings, DEFAULT_NOTIFICATION_SETTINGS)) {
-            Set<NotificationSetting> updatedDefaultNotificationSettings = new HashSet<>(DEFAULT_NOTIFICATION_SETTINGS);
 
-            userNotificationSettings.forEach(userNotificationSetting -> DEFAULT_NOTIFICATION_SETTINGS.forEach(defaultSetting -> {
-                if (defaultSetting.getSettingId().equals(userNotificationSetting.getSettingId())) {
-                    updatedDefaultNotificationSettings.remove(defaultSetting);
-                    updatedDefaultNotificationSettings.add(userNotificationSetting);
-                }
-            }));
-
-            updatedDefaultNotificationSettings.forEach(userNotificationSetting -> userNotificationSetting.setUser(user));
-            // update DB to fix inconsistencies and avoid redundant future merges
-            // first remove all settings of the current user in the DB
-            notificationSettingRepository.deleteAll(userNotificationSettings);
-            // save correct merge to DB
-            notificationSettingRepository.saveAll(updatedDefaultNotificationSettings);
-            return updatedDefaultNotificationSettings;
+        // If all default settings are available in the database, return them
+        if (haveSameNotificationSettingIds(userNotificationSettings, defaultSettings)) {
+            return userNotificationSettings;
         }
-        return userNotificationSettings;
+
+        // Merge user settings with default settings
+        Map<String, NotificationSetting> settingsMap = userNotificationSettings.stream().collect(Collectors.toMap(NotificationSetting::getSettingId, setting -> setting));
+
+        // Merge user-specific settings with default settings:
+        // - If the user already has a setting with the same ID, use the user’s version.
+        // - If the user doesn’t have a specific setting, fall back to the default version.
+        Set<NotificationSetting> mergedSettings = defaultSettings.stream().map(defaultSetting -> settingsMap.getOrDefault(defaultSetting.getSettingId(), defaultSetting))
+                .collect(Collectors.toSet());
```

Comment:
I think like this it would be easier to understand for me (adjusted the comments a little) 
```suggestion
        Map<String, NotificationSetting> settingsMap = userNotificationSettings.stream().collect(Collectors.toMap(NotificationSetting::getSettingId, setting -> setting));

        // defaultSettings might have changed (e.g. number of settings) -> need to merge the saved settings with default ones (else errors appear)
        //
        // Merge user-specific settings with default settings:
        // - If the user already has a setting with the same ID, use the user’s version.
        // - If the user doesn’t have a specific setting, fall back to the default version.
        Set<NotificationSetting> mergedSettings = defaultSettings.stream().map(defaultSetting -> settingsMap.getOrDefault(defaultSetting.getSettingId(), defaultSetting))
                .collect(Collectors.toSet());
```